python/setup.py

```diff
@@ -27,7 +27,7 @@
         "requests<3.0.0",
         "numpy<2.0.0",
         "flatbuffers<2.0.0",
-        "protobuf<4.0.0",
+        "protobuf==3.20.3",
```

Comment:
That will put hard constraint on `protobuf` version in the user environment causing potential conflicts with other packages.

Reply:
Are both the change in the `protobuf` version and the change in `importlib_metadata` required? I tried locally without those and seems the main Python tests are happy (haven't tried with anything else though).